lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -3130,7 +3130,7 @@ void ByteCodeGenerator::ProcessCapturedSym(Symbol *sym)
     FuncInfo *funcHome = sym->GetScope()->GetFunc();
     FuncInfo *funcChild = funcHome->GetCurrentChildFunction();
 
-    Assert(sym->NeedsSlotAlloc(funcHome) || sym->GetIsGlobal() || sym->GetIsModuleImport());
+    Assert(sym->NeedsSlotAlloc(funcHome) || sym->GetIsGlobal() || sym->GetIsModuleImport() || sym->GetIsModuleExportStorage());
```

Comment:
No, symbols being module import or export is perpendicular to initializing non-temp register on block entry.

---
In reply to: [120192836](https://github.com/Microsoft/ChakraCore/pull/3079#discussion_r120192836) [](ancestors = 120192836)